onnx/defs/math/defs.cc

```diff
@@ -2997,40 +2995,52 @@ ONNX_OPERATOR_SET_SCHEMA(
+            "If `axis=N` and `onesided` is `1`, the following shape is expected: `[signal_dim0][signal_dim1][signal_dim2]...[floor(signal_dimN/2)+1][2]`. "
+            "The `signal_dim` at the specified `axis` is equal to the `dft_length`.",
             "T1")
-        .TypeConstraint(
-            "T1",
-            {"tensor(float16)", "tensor(float)", "tensor(double)", "tensor(bfloat16)"},
-            "Constrain input and output types to float tensors.")
-        .TypeConstraint("T2", {"tensor(int32)", "tensor(int64)"}, "Constrain scalar length types to int64_t.")
+        .TypeConstraint("T1", OpSchema::all_float_types_ir4(), "Constrain input and output types to float tensors.")
+        .TypeConstraint("T2", {"tensor(int32)", "tensor(int64)"}, "Constrain scalar length types to integers.")
         .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
           bool is_onesided = static_cast<bool>(getAttribute(ctx, "onesided", 0));
           bool inverse = static_cast<bool>(getAttribute(ctx, "inverse", 0));
 
+          const size_t input_arg_index = 0;
+          const size_t axis_arg_index = 1;
+          const size_t dft_length_arg_index = 2;
+          const size_t output_index = 0;
+
           if (inverse && is_onesided) {
             fail_shape_inference("is_onesided and inverse attributes cannot be enabled at the same time");
           }
 
-          propagateElemTypeFromInputToOutput(ctx, 0, 0);
-          if (!hasInputShape(ctx, 0)) {
+          propagateElemTypeFromInputToOutput(ctx, input_arg_index, output_index);
+          if (!hasInputShape(ctx, input_arg_index)) {
             // If no shape is available for the input, skip shape inference...
             return;
           }
 
           // In general the output shape will match the input shape exactly
           // So initialize the output shape with the input shape
-          auto& input_shape = getInputShape(ctx, 0);
+          auto& input_shape = getInputShape(ctx, input_arg_index);
           ONNX_NAMESPACE::TensorShapeProto result_shape_proto = input_shape;
 
           // Get the axis where the DFT will be performed.
-          auto axis = static_cast<int>(getAttribute(ctx, "axis", 1));
-          auto rank = input_shape.dim_size();
+          const TensorProto* axis_tensor = ctx.getInputData(axis_arg_index);
+          if (axis_tensor == nullptr) {
+            // Skip if axis is not known
+            return;
+          }
+          // TODO(justinchuby): Create invariance checking functions to ensure shapes and sizes
+          // to abstrct the following logic out.
+          if (axis_tensor->dims_size() != 0) {
+            fail_shape_inference("axis input must be a scalar.");
+          }
+          const int64_t axis = ONNX_NAMESPACE::defs::math::utils::GetScalarValueFromTensor<int64_t>(axis_tensor);
```

Comment:
The error `undefined symbol: _ZN4onnx4defs4math5utils24GetScalarValueFromTensorIlEET_PKNS_11TensorProtoE` seems related to naming space of GetScalarValueFromTensor. Have you tried to use `GetScalarValueFromTensor` directly?